src/components/LibrariesList.tsx

```diff
@@ -16,26 +17,53 @@ export interface LibrariesListOwnProps {
 
 export interface LibrariesListDispatchProps {
   fetchData: () => Promise<LibrariesData>;
+  search: (data: FormData) => Promise<LibrariesData>;
+}
+
+export interface LibrariesListState {
+  showAll: boolean;
+  searchTerm: string;
 }
 
 export interface LibrariesListProps extends LibrariesListStateProps, LibrariesListOwnProps, LibrariesListDispatchProps {};
 
-export class LibrariesList extends React.Component<LibrariesListProps, {}> {
+export class LibrariesList extends React.Component<LibrariesListProps, LibrariesListState> {
+  constructor(props: LibrariesListProps) {
+    super(props);
+    this.search = this.search.bind(this);
+    this.clear = this.clear.bind(this);
+    this.state = { showAll: true, searchTerm: "" };
+  }
 
   render(): JSX.Element {
+    let libraryItems;
     let hasLibraries = (this.props.libraries && this.props.libraries.libraries && this.props.libraries.libraries.length > 0);
-    return(
+    let libraries = hasLibraries && this.props.libraries.libraries;
+    if (hasLibraries) {
+      libraryItems = libraries.map(library =>
+        <LibrariesListItem
+          key={library.uuid}
+          library={library}
+          store={this.props.store}
+        />
+      );
+    } else {
+        libraryItems = !this.state.searchTerm.length && <span className="page-header">There are no libraries in this registry yet.</span>;
+    }
+
+    let form = (<SearchForm
+      search={this.search}
+      term={this.state.searchTerm}
+      text="Search for a library by name"
+      inputName="name"
+      clear={!this.state.showAll ? this.clear : null}
+      resultsCount={libraries && libraries.length}
+    />);
+
+    return (
       <ul className="list">
-        { hasLibraries ?
-          this.props.libraries.libraries.map(library =>
-            <LibrariesListItem
-              key={library.uuid}
-              library={library}
-              store={this.props.store}
-            />
-          ) :
-          <span className="page-header">There are no libraries in this registry yet.</span>
-        }
+        {form}
```

Comment:
A couple of semantic updates here: the `SearchForm` isn't a list item but a `Panel`. It should not belong in the `ul`, and maybe the error message about no libraries in the list should also be a `li` since the parent is a `ul`.

Reply:
The library `Panel`s themselves in the ul aren't list items so maybe wrap them around an `li`.